app/customer.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+from typing import List
+import math
+
+from app.car import Car
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(
+            self,
+            name: str,
+            product_cart: dict,
+            location: list,
+            money: float | int,
+            car: dict
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.location = location
+        self.money = money
+        self.car = Car(**car)
+
+    def shopping(self, shops: List[Shop], fuel_price: float | int) -> None:
+        self.__check_money()
+        self.get_shop(shops, fuel_price)
+
+    def __calculate_cost_trip(
+            self,
+            fuel_price: float | int,
+            distance: int | float
+    ) -> float:
+        return round(
+            self.car.fuel_consumption / 100 * distance * fuel_price * 2,
+            2
+        )
+
+    def get_shop(self, shops: List[Shop], fuel_price: int | float) -> None:
+        self_x, self_y = self.location
+        best_shop = {"total_cost": None, "shop": None}
+
+        for shop in shops:
+            total_cost = 0
+            total_shop = best_shop["total_cost"]
+            shop_x, shop_y = shop.location
+            distance = math.sqrt(
```

Comment:
distance = math.sqrt(
                (self_x - shop_x) ** 2 + (self_y - shop_y) ** 2
            ) better write in a separate function.